### DISCUSSION TITLE
A bit clueless about returned error message

### DISCUSSION THREAD

**Discussion by Elkaid** [mkaring/ConfuserEx#352] [Q&A]
Hello everyone,

I'm getting an error message "Constant is NULL"
Everything has been working fine for several years.  
I have tested several past releases and still getting the same behavior.

![Capture](https://user-images.githubusercontent.com/32937890/118520788-5ba39280-b708-11eb-976b-782c520be3c9.PNG)

I have absolutely no clue what could be causing that.  I tried disabling constant protection and other modules without any results.

Thanks for your help

**Comment by mkaring** [comment]
This is the first time I have seen an error like this. It looks like it was expecting a constant value (like a string or an array initialization) it has to store in the meta data of the assembly, but fails because there is no data.
Could you try running ConfuserEx with that assembly, but disable all the protections. This should show if the protections are breaking something or if the assembly contains something strange to start with.

**Reply by Elkaid** [reply]
Thank you for your reply,

I tried isolating a culprit by trying one by one each protection but no changes.  Still no joy when using the following project

![Capture2](https://user-images.githubusercontent.com/32937890/118655370-7d0f8780-b7b7-11eb-88d5-59eb89715375.PNG)

The problem seems to appear with several different ConfuserEx-based Projects/Branches

Thanks for your help

**Reply by mkaring** [reply]
This means there is something in your assembly that is causing the failure, even before ConfuserEx is touching it. Sadly, without the assembly it will be extremely difficult to identify what the actual problem is. Do you have a known working version of the assembly so you could potentially isolate the change to the source code that is causing the problem?

The only alternative I can offer is for you to send that assembly over to me privately, so I can analyze what is causing the problem.

**Reply by Elkaid** [reply]
Thank you very much for your reply,

I'll do some investigation and get back to you if I find anything.

Regards

Emmanuel
